sensor/common/registry/registry_store.go

```diff
@@ -201,3 +213,68 @@ func (rs *Store) GetGlobalRegistryForImage(image *storage.ImageName) (registryTy
 
 	return nil, errors.Errorf("unknown image registry: %q", reg)
 }
+
+// SetDelegatedRegistryConfig sets a new delegated registry config for use in determining
+// if a particular image is from a registry that should be accessed local to this cluster.
+func (rs *Store) SetDelegatedRegistryConfig(config *central.DelegatedRegistryConfig) {
+	rs.delegatedRegistryConfigMutex.Lock()
+	defer rs.delegatedRegistryConfigMutex.Unlock()
+	rs.delegatedRegistryConfig = config
+}
+
+// IsLocal determines if an image is from a registry that should be accessed
+// local to this secured cluster.  Always returns true for image registries that have
+// been added via AddClusterLocalRegistryHost.
+func (rs *Store) IsLocal(image *storage.ImageName) bool {
+	if image == nil {
+		return false
+	}
+
+	if rs.hasClusterLocalRegistryHost(image.GetRegistry()) {
+		// This host is always cluster local irregardless of the DelegatedRegistryConfig (ie: OCP internal registry).
+		return true
+	}
+
+	rs.delegatedRegistryConfigMutex.RLock()
+	defer rs.delegatedRegistryConfigMutex.RUnlock()
+
+	config := rs.delegatedRegistryConfig
+	if config == nil || config.EnabledFor == central.DelegatedRegistryConfig_NONE {
+		return false
+	}
+
+	if config.EnabledFor == central.DelegatedRegistryConfig_ALL {
+		return true
+	}
+
+	// if image matches a delegated registry prefix, it is local
+	imageFullName := urlfmt.TrimHTTPPrefixes(image.GetFullName())
+	for _, r := range config.Registries {
```

Comment:
do we ever expect this to be a large number of registries?

Reply:
I don't expect it be large, anticipating most entries to be at the 'host' level or one layer lower (ie: `example.com/dev`, `example.com/prod`).